demo.py

```diff
@@ -121,19 +121,19 @@ def build_cqm(G, k):
 
     # Add binary variables, one for each node and each partition in the graph
     print("\nAdding variables....")
-    v = [[Binary(f'v_{i},{k}') for k in partitions] for i in G.nodes]
+    v = [[Binary(f'v_{i},{p}') for p in partitions] for i in G.nodes]
 
     # One-hot constraint: each node is assigned to exactly one partition
     print("\nAdding one-hot constraints...")
     for i in G.nodes:
         # print("\nAdding one-hot for node", i)
-        cqm.add_discrete([f'v_{i},{k}' for k in partitions], label=f"one-hot-node-{i}")
+        cqm.add_discrete([f'v_{i},{p}' for p in partitions], label=f"one-hot-node-{i}")
 
     # Constraint: Partitions have equal size
     print("\nAdding partition size constraint...")
     for p in partitions:
         # print("\nAdding partition size constraint for partition", p)
-        cqm.add_constraint(quicksum(v[n][p] for n in G.nodes) == G.number_of_nodes()/k, label='partition-size-{}'.format(p))
+        cqm.add_constraint(quicksum(v[i][p] for i in G.nodes) == G.number_of_nodes()/k, label='partition-size-{}'.format(p))
```

Comment:
While we're clarifying, perhaps
```suggestion
        cqm.add_constraint(quicksum(v[i][p] for i in G.nodes) == G.number_of_nodes()/k, label=f'partition-size-{p}')
```
as well?

Reply:
updated!